include/albatross/src/covariance_functions/covariance_function.hpp

```diff
@@ -307,26 +307,26 @@ class SumOfCovarianceFunctions
   }
 
   template <typename X,
-  typename std::enable_if<has_valid_ssr_impl<LHS, X>::value &&
-                          has_valid_ssr_impl<RHS, X>::value,
-                                      int>::type = 0>
+            typename std::enable_if<has_valid_ssr_impl<LHS, X>::value &&
+                                        has_valid_ssr_impl<RHS, X>::value,
+                                    int>::type = 0>
   auto _ssr_impl(const std::vector<X> &xs) const {
     return concatenate(this->lhs_.state_space_representation(xs),
-        this->rhs_.state_space_representation(xs));
+                       this->rhs_.state_space_representation(xs));
```

Comment:
I can't help but feel like concatenation is the wrong operation.
Imagine adding two squared exponentials of different length_scale. They'll have the same range, but the number of inducing points coming from each of them will depend on the length_scale. You could get duplicate inducing points (midpoint?) or ones in close proximity.

Reply:
Yeah, not bullet proof yet.  I would prefer to not pass around `std::set` since everywhere else uses `std::vector` ... but maybe it's more appropriate.  Alternatively we could filter out all duplicates in the `state_space_representation` method, but that would result a lot of repeated work.  For now I think I'm OK leaving it up to the user to realize there are duplicates and decide if that's an issue.